rust/lance-table/src/io/commit.rs

```diff
@@ -118,16 +142,61 @@ async fn current_manifest_path(object_store: &dyn ObjectStore, base: &Path) -> R
     }
 }
 
+// This is an optimized function that searches for the latest manifest. In
+// object_store, list operations lookup metadata for each file listed. This
+// method only gets the metadata for the found latest manifest.
+fn current_manifest_local(base: &Path) -> std::io::Result<Option<ManifestLocation>> {
```

Comment:
just curious, why we can't get the latest manifest in `/path/to/dataset/latest.manifest`

Reply:
We no longer use `latest.manifest`. It was a very flawed design, given mutable files and object storage don't mix well. This explained in detail in #1362, #1365.

Reply:
gotcha, thanks, but I see some documents in our codebase still describe `latest.manifest` though, like in `docs/format/rst`